File: platforms/kotlin/examples/core/renderer/load.kt
import org.fragmentcolor.*

val renderer = Renderer()
val target = renderer.createTextureTarget(64u, 64u)

val mesh = Mesh()
mesh.addVertex( Vertex.pbr(listOf(0.0f, 0.5f, 0.0f)).set("uv0", floatArrayOf(0.5f, 1.0f)), )
val bytes: ByteArray = byteArrayOf()
// Raw 2×2 RGBA pixel bytes — uploaded lazily by """Renderer.load""" below.
 
// vocabulary covers all of them.
val red_pixels = listOf(255.0f, 0.0f, 0.0f, 255.0f, 0.0f, 255.0f, 0.0f, 255.0f, 0.0f, 0.0f, 255.0f, 255.0f, 255.0f, 255.0f, 255.0f, 255.0f, .0f)
val red_tex = renderer.createTexture(TextureInputMobile.Bytes(red_pixels.let { ba -> ByteArray(ba.size) { i -> ba[i].toInt().and(0xFF).toByte() } }), null)
val material = Material.pbr().baseColorTexture(red_tex)
val model = Model(mesh, material)
val scene = Scene()
scene.add(model)

// Eager prewarm — uploads the pending texture(s) so the next render is
// GPU-only.
renderer.load(scene)
renderer.render(scene, target)

File: platforms/web/examples/core/renderer/load.js
import { Material, Mesh, Model, Renderer, Scene, Vertex } from "fragmentcolor";

const renderer = new Renderer();
const target = await renderer.createTextureTarget([64, 64]);

const mesh = new Mesh();
mesh.addVertex( Vertex.new([0.0, 0.5, 0.0]) .set(Vertex.NORMAL, [0.0, 0.0, 1.0]) .set(Vertex.UV0, [0.5, 1.0]), );
// Raw 2×2 RGBA pixel bytes — uploaded lazily by `Renderer.load` below.
// In practice the loader hands the setter encoded PNG/JPEG bytes (from a
// BIN chunk) or a file path (from a URI); the same `Into<TextureInput>`
// vocabulary covers all of them.
const red_pixels = [ 255,   0,   0, 255,    0, 255,   0, 255, 0,   0, 255, 255,  255, 255, 255, 255, ];
const material = Material.pbr()?.baseColorTexture(red_pixels, [2, 2]);
const model = new Model(mesh, material);
const scene = new Scene();
scene.add(model);

// Eager prewarm — uploads the pending texture(s) so the next render is
// GPU-only.
await renderer.load(scene);
renderer.render(scene, target);